src/test/java/frc/robot/subsystems/gpm/ShooterTest.java

```diff
@@ -26,20 +32,63 @@ public void motorValueTest() {
     /**
      * Check that the conversion routines are inverses of each other.
      */
-    // @Test
-    // public void conversionTest() {
-    //     // a test rpm
-    //     double rpmInput = 1500.0;
-    //     // converted to a velocity
-    //     double velocity = Shooter.shooterRPMToSpeed(rpmInput);
+    @Test
+    public void conversionTest() {
+        // a test rpm
+        double rpmInput = 1500.0;
+        // converted to a velocity
+        double velocity = Shooter.shooterRPMToSpeed(rpmInput);
         
-    //     // converting velocity back to an rpm should match the input
-    //     assertEquals(rpmInput, Shooter.shooterSpeedToRPM(velocity), 0.0001);
-
-    //     // 6000 rpm is 100 rps. circumference is pi * 4 inches
-    //     assertEquals(
-    //         100.0 * Math.PI * Units.inchesToMeters(4.0),
-    //         Shooter.shooterRPMToSpeed(6000.0),
-    //         0.0001);
-    // }
+        // converting velocity back to an rpm should match the input
+        assertEquals(rpmInput, Shooter.shooterSpeedToRPM(velocity), 0.0001);
+
+        // 6000 rpm is 100 rps. circumference is pi * 4 inches. Geared up by 2.0.
+        assertEquals(
+            (1.0 / Shooter.gearRatio) * 100.0 * Math.PI * Units.inchesToMeters(4.0),
+            Shooter.shooterRPMToSpeed(6000.0),
+            0.0001);
+    }
+
+    @Test
+    public void spinUpTest() {
+        // shooter RPM should be at zero
+        assertEquals(0.0, shooter.getRightMotorRPM(), 0.0001);
+        assertEquals(0.0, shooter.getLeftMotorRPM(), 0.00001);
+
+        // set the target speeds to zero
+        shooter.setTargetRPM(0.0, 0.0);
+
+        // shooter PID should be satisfied
+        // satisfied is cached? Do some calculations to fill the cache.
+        shooter.leftPID.calculate(0.0);
+        shooter.rightPID.calculate(0.0);
+        assertTrue(shooter.atSetpoint());
+
+        // set the motor speed to 1000 RPM
+        shooter.setTargetRPM(1000.0, 1000.0);
+        // we should not be satisfied
+        assertFalse(shooter.atSetpoint());
+
+        // run some simulations
+        // TODO: after 3 seconds, we are still 75 RPM away!
+        // Tolerance is 80 RPM, so we are at the set point.
+        for (int i = 0; i < 150; i++) {
+            shooter.simulationPeriodic();
+            shooter.periodic();
+            System.out.println(shooter.getLeftMotorRPM());
```

Comment:
Can you only print the last frame so we don't have to scroll through 150 prints to read the other tests?